src/gui/ClipView.cpp

```diff
@@ -463,30 +461,30 @@ void TrackContentObjectView::dropEvent( QDropEvent * de )
 	if( m_trackView->trackContainerView()->allowRubberband() == true )
 	{
 		TrackContentWidget * tcw = getTrackView()->getTrackContentWidget();
-		TimePos tcoPos = TimePos( m_tco->startPosition() );
+		TimePos clipPos = TimePos( m_clip->startPosition() );
 
-		if( tcw->pasteSelection( tcoPos, de ) == true )
+		if( tcw->pasteSelection( clipPos, de ) == true )
 		{
 			de->accept();
 		}
 		return;
 	}
 
-	// Don't allow pasting a tco into itself.
+	// Don't allow pasting a clip into itself.
 	QObject* qwSource = de->source();
 	if( qwSource != nullptr &&
-	    dynamic_cast<TrackContentObjectView *>( qwSource ) == this )
+	    dynamic_cast<ClipView *>( qwSource ) == this )
 	{
 		return;
 	}
 
-	// Copy state into existing tco
+	// Copy state into existing clip
 	DataFile dataFile( value.toUtf8() );
-	TimePos pos = m_tco->startPosition();
-	QDomElement tcos = dataFile.content().firstChildElement( "tcos" );
-	m_tco->restoreState( tcos.firstChildElement().firstChildElement() );
-	m_tco->movePosition( pos );
-	AutomationPattern::resolveAllIDs();
+	TimePos pos = m_clip->startPosition();
+	QDomElement clips = dataFile.content().firstChildElement( "tcos" ); //TODO: rename "tcos" to "clips"
```

Comment:
Any reason `tcos` was kept here?

Reply:
Because the PR intends to not change anything in the Datafile?

Reply:
This is just a dummy datafile tho, for clipboard and drag-n-drop, should be safe... [here's a better link](https://github.com/LMMS/lmms/blob/55d361fb6513193c465be1924167fb179c523fd3/src/gui/ClipView.cpp#L484)